File: src/at.rb
require 'quickfix_ruby'
require '../src/at_application'

if( ARGV.length == 0 )
	puts "usage: at.rb -f FILE"
	exit 1
end

file = ARGV[0]

settings = Quickfix::SessionSettings.new( file )
application = Application.new
factory = Quickfix::FileStoreFactory.new( "store" )

acceptor = Quickfix::SocketAcceptor.new( application, factory, settings )

acceptor.start()
while( true )
	sleep( 1 )
end
acceptor.stop()

File: src/at.py
from __future__ import print_function
import sys
import time
import quickfix as fix
import at_application

if len(sys.argv) == 0:
	print("usage: at.py -f FILE")
	exit

file = sys.argv[1]

settings = fix.SessionSettings( file )
application = at_application.Application()
factory = fix.FileStoreFactory( "store" )

acceptor = fix.SocketAcceptor( application, factory, settings )

acceptor.start()
while 1:
	time.sleep( 1 )
acceptor.stop()
